File: src/nodes/continue.cpp
#include "nodes/continue.hpp"
#include "core/emitter.hpp"
#include <iostream>

using namespace nodes;

NodeContinue::NodeContinue(const utils::Position &position)
    : Node(position) {

}

void NodeContinue::show(std::string &indent, char end) const {
    std::cout << indent << "NodeContinue" << end;
}

NodeKind NodeContinue::getKind() const {
    return NodeKind::Continue;
}

void NodeContinue::emit(core::Emitter &emitter) const {
    emitter.writeU64(common::bytecodes::ApicaBytecode::Continue);
}

void NodeContinue::setId() {
    
}

utils::OptimizedResult NodeContinue::optimize(core::Optimizer &) {
    return utils::OptimizedResult(utils::OptimizedFlag::AlwaysNull);
}

File: src/nodes/_continue.rs
use crate::utils::position::Position;

pub struct NodeContinue {
    position: Position,
}

impl NodeContinue {
    pub fn init(position: Position) -> NodeContinue {
        NodeContinue { position }
    }
    
    pub fn get_position(&self) -> &Position {
        &self.position
    }
    
    pub fn show(&self, indent: &mut String, end: char) {
        print!("{indent}NodeContinue{end}");
    }
}